android/titanium/src/java/org/appcelerator/titanium/util/TiDownloadManager.java

```diff
@@ -146,13 +146,7 @@ public void run()
 
```

Comment:
The above `if (TiApplication.isUIThread()) { ... }` code block should be changed to throw an exception too for consistency. This code block can be tested when setting a Titanium view's `backgroundImage` property.

How about something like this?
```java
// If running on main UI thread, then block this thread while doing download on another thread.
// Note: Using "HttpUrlConnection" on UI thread will cause a "NetworkOnMainThreadException" to be thrown.
if (TiApplication.isUIThread()) {
	// Provides the error result of the threaded-out runnable below.
	class RunnableResult
	{
		Exception exception;
	}

	// Perform the blocking download on another thread.
	// Downloaded content will be made available via Titanium's "TiResponseCache".
	final RunnableResult runnableResult = new RunnableResult();
	Thread thread = new Thread(new Runnable() {
		@Override
		public void run()
		{
			try (InputStream stream = blockingDownload(uri)) {
				if (stream != null) {
					KrollStreamHelper.pump(stream, null);
				}
			} catch (Exception ex) {
				runnableResult.exception = ex;
			}
		}
	});
	thread.start();
	thread.join();
	if (runnableResult.exception != null) {
		throw runnableResult.exception;
	}

	// Return a stream to the downloaded file/content via our response cache.
	URI cachedUri = TiResponseCache.fetchEndpointFollowingRedirects(uri);
	if (cachedUri != null) {
		inputStream = TiResponseCache.openCachedStream(cachedUri);
	}
	return inputStream;
}
```